app/tool/tasks.cxx

```diff
@@ -252,6 +252,191 @@ static void gen_scan(Target* tgt) {
 }
 
 /**
+ * TASKS.TRIM_INV_SCAN
+ * 
+ * Scan TRIM_INV Parameter across all 63 parameter points for each channel
+ * Check ADC Pedstals for each parameter point
+ * Used to trim pedestals within each link
+ */
+static void trim_inv_scan(Target* tgt) {
+  int nevents = pftool::readline_int("Number of events per point: ", 1);
+  int ch_start = pftool::readline_int("Start channel (m): ", 0);
+  int ch_end   = pftool::readline_int("End channel (n): ", 71);
+  std::vector<std::string> trigger_types = { "PEDESTAL", "CHARGE" };
+  std::string trigger = pftool::readline("Trigger type: ", trigger_types);
+
+  std::string output_filepath = pftool::readline_path("trim_inv_scan", ".csv");
+
+  auto roc = tgt->hcal().roc(pftool::state.iroc, pftool::state.type_version());
+
+  boost::json::object header;
+  header["scan_type"] = "CH_#.TRIM_INV sweep";
+  header["trigger"] = trigger;
+  header["nevents_per_point"] = nevents;
+  header["channel_range"] = std::to_string(ch_start) + "-" + std::to_string(ch_end);
+
+  pflib::DecodeAndWriteToCSV writer{
+    output_filepath,
+    [&](std::ofstream& f) {
+      f << std::boolalpha
+        << "# " << boost::json::serialize(header) << '\n';
+      f << "channel,TRIM_INV," << pflib::packing::Sample::to_csv_header << '\n';
+    },
+    [&](std::ofstream& f, const pflib::packing::SingleROCEventPacket& ep) {
+      f << header["current_channel"].as_int64() << ',' << header["current_trim_inv"].as_int64() << ',';
+      ep.channel(header["current_channel"].as_int64()).to_csv(f);
+      f << '\n';
+    }
+  };
+
+  tgt->setup_run(1 /* dummy */, DAQ_FORMAT_SIMPLEROC, 1 /* dummy */);
+
+  for (int ch = ch_start; ch <= ch_end; ++ch) {
+    pflib_log(info) << "Running CH_" << ch;
```

Comment:
Similar to what I mentioned in the other comment. You don't need to do separate runs for the different channels unless you are worried about electronic cross talk. (We do not expect this to be an issue for pedestals I believe).

Instead of doing different runs, you would just apply all of the testing TRIM_INV to all the channels at once. e.g.
```cpp
auto trim_inv_test_builder = roc.testParameters();
for (int ch{0}; ch < 72; ch++) {
  trim_inv_test_buildr.add("CH_"+std::to_string(ch), "TRIM_INV", trim);
}
auto trim_inv_test = trim_inv_test_builder.apply();
```